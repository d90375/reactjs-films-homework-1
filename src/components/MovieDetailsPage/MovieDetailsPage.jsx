import React from 'react';
import MovieDetailsInfo from '../MovieDetailsInfo';
import MovieDetailsDescription from '../MovieDetailsDescription';
import MovieDetailsRating from '../MovieDetailsRating';
import MovieDetailsButtons from '../MovieDetailsButtons';

import styles from './movieDetailsPage.scss';

const data = {
  title: 'The Jungle Book',
  genres: ['Adventure', 'Drama', 'Family', 'Fantasy'],
  duration: '1h 46m',
  description: `There are growing dangers in the wizarding world of 1926 New York.  
                Something mysterious is leaving a path of destruction in the streets, 
                threatening to expose the wizarding`,
  rating: '4.8',
};

const MovieDetailsPage = () => {
  const { description, rating, ...info } = data;

  return (
    <section className={styles.movieDetailsPage}>
      <div className={styles.info}>
        <MovieDetailsInfo info={info} />
        <MovieDetailsDescription description={description} />
      </div>
      <div className={styles.info}>
        <MovieDetailsRating rating={rating} />
        <MovieDetailsButtons />
      </div>

    </section>
  );
};

export default MovieDetailsPage;
