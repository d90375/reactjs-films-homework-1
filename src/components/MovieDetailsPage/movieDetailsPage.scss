.movieDetailsPage {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    height: 45vw;
    max-height: 1100px;
    padding: 0 5%;
    background-color: rgb(0, 0, 0);
    background-image: url('assets/images/junglebook.png');
    background-size: cover; 
}

.info {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
}
