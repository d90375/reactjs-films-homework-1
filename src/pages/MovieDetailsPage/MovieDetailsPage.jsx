import React from 'react';
import Header from '../../containers/Header';
import MovieDetails from '../../containers/MovieDetails';
import MoviesSection from '../../containers/MoviesSection';
import Footer from '../../components/Footer';

const MainPage = () => (
  <div>
    <Header />
    <MovieDetails />
    <MoviesSection />
    <Footer />
  </div>
);

export default MainPage;
