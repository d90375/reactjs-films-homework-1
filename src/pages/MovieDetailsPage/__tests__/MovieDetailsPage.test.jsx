import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieDetailsPage from '../MovieDetailsPage';

test('MovieDetailsPage. renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MovieDetailsPage />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
