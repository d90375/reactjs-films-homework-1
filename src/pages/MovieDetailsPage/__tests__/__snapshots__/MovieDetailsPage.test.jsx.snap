// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsPage. renders correctly 1`] = `
<div>
  <Memo(Connect(withRouter(Header))) />
  <Memo(Connect(MovieDetails)) />
  <Memo(Connect(withRouter(MoviesSection))) />
  <Footer />
</div>
`;
